e2e/utils/setup_nns.bash

```diff
@@ -0,0 +1,66 @@
+#! /bin/bash
+
+# The commit hash was get from ic release note
+IC_COMMIT="3b5d893c0857c47715fc339112e5dd1dbfff77a8"
+
+if test -z "${NNS_ARTIFACTS}"
+then
+  NNS_ARTIFACTS=$(mktemp -d -t dfx-e2e-nns-artifacts-XXXX) # Linux mktemp requires Xs in suffix
```

Comment:
Can we clean these up after the tests run?  It looks like the intent here is to download the artifacts once per invocation of ledger.bash, rather than once per test (which is why the artifacts aren't stored under DFXE2E_TEMP_DIR, created in standard_setup and cleaned up in standard_teardown).

It's convenient that the test automatically downloads the artifacts that it needs, but this also means that every time someone runs `bats e2e/tests-dfx/ledger.bash`,  it downloads 10 wasm files an the ic-nns-init binary, and leaves them in a temp dir.

Overall I'd suggest calling an equivalent of this script from the provision scripts, and exporting E2E_NNS_ARTIFACTS with the location of the artifacts as a precondition for running these tests.

Reply:
Thx! I'll move them into provision scripts.